lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,309 @@
+    except:
+        return False
+    else:
+        return True
+
+
+def increment_is_valid(nn_taxon):
+    # check the increment is an int and has at least 2 digits
+    if len(nn_taxon) >= 2:
+        try:
+            int(nn_taxon)
+        except 0:
+            return False
+
+        else:
+            return True
+    else:
+        return False
+
+
+def starts_alphanumeric(taxon):
+    # check the element starts with a letter or number 
+    match = re.search('^[a-zA-Z0-9]', taxon)
+    if match:
+        return True
+    else:
+        return False
+
+
+def validate(user_input):
+    if ('.' in user_input) or (len(user_input) > 60):
+        return False
+
+    name = [x.strip() for x in user_input.upper().split(':')]
+
+    for element in name:
+        if not starts_alphanumeric(element):
+            print('Invalid')
+            return False
+
+    # Check the length of the name
+    if (len(name) < 2) or (len(name) > 5):
+        print('Invalid')
+        return False
+
+    # Validate the functional component (FFFFF)
+    fc_taxon = str(name[0])
+    fc_valid = functional_component_is_valid(fc_taxon)
+
+    second_element = name[1]
+
+    if fc_valid:
+        # Check for name with 2 elements (FFFFF:CCC)
+        if (len(name)==2):
+            ccc_valid = constituent_component_is_valid(second_element)
+
+            if ccc_valid:
+                print('Valid')
+                return True
+            else:
+                print('Invalid')
+                return False
+
+        third_element = name[2]
+
+        # check for name with 3 elements (FFFFF:GGG:CCC, FFFFF:CCC:NN, FFFFF:CCC:XXXX)
+        if (len(name) == 3):
+            fg_valid = fungible_is_valid(second_element)
+            ccc_valid = constituent_component_is_valid(second_element)
+
+            if fg_valid:   
+                ccc_valid = constituent_component_is_valid(third_element)
+                if ccc_valid:
+                    print('Valid')
+                    return True
+                else:
+                    print('Invalid')
+                    return False
+            elif ccc_valid:
+                    element_valid = starts_alphanumeric(third_element)
+                    if element_valid:
+                        print('Valid')
+                        return True
+                    else:
+                        print('Invalid')
+                        return False
+            else:
+                print('Invalid')
+                return False
+
+        fourth_element = name[3]
+
+        # Check for name with 4 elements (FFFFF:GGG:CCC:NN, FFFFF:GGG:CCC:XXXX, FFFFF:CCC:NN:XXXX)
+        if (len(name) == 4):
+            fg_valid = fungible_is_valid(second_element)
+            ccc_valid = constituent_component_is_valid(second_element)
+
+            if fg_valid:
+                ccc_valid = constituent_component_is_valid(third_element)
+                fourth_element_valid = starts_alphanumeric(fourth_element)
```

Comment:
This is another case where a :NN is included in the above comment but will be marked invalid